controllers/awscluster_controller.go

```diff
@@ -135,6 +135,9 @@ func reconcileNormal(clusterScope *scope.ClusterScope) (reconcile.Result, error)
 	// If the AWSCluster doesn't have our finalizer, add it.
 	if !util.Contains(awsCluster.Finalizers, infrav1.ClusterFinalizer) {
 		awsCluster.Finalizers = append(awsCluster.Finalizers, infrav1.ClusterFinalizer)
+		if err := clusterScope.Update(); err != nil {
```

Comment:
Why do we need to call this here? When this function returns, the caller calls `Close()` on defer as well

Reply:
Unless I'm mistaken, because the AWS calls take a long time (seconds to minutes) the finalizer isn't yet persisted and you can delete the AWS cluster during this time. This immediately persists the finalizer before the longer running AWS calls.

Reply:
@vincepri also I opened #1454 that describes the issue

Reply:
Makes sense, let's add a comment to that effect

Reply:
@vincepri comment added. Feel free to wordsmith it differently if it's not clear.